### ISSUE TITLE
Improve login handling

### ISSUE THREAD

**Issue by eddybl** [N4S4/synology-api#106]
I am just starting with the API and tried to connect using the core module.

For me, `base_api_core.Core()` [failed on line 9](https://github.com/N4S4/synology-api/blob/3f7feaf0f265ddf15e6a7008d738baafbe016db2/synology_api/base_api_core.py#L9):

```python
self.session.login('Core')
```

Relevant part of the traceback:

```python
<...>synology_api\base_api_core.py in __init__(self, ip_address, port, username, password, secure, cert_verify, dsm_version, debug, otp_code)
      7         self.session = syn.Authentication(ip_address, port, username, password, secure, cert_verify, dsm_version, debug, otp_code)
      8 
----> 9         self.session.login('Core')
     10         self.session.get_api_list('Core')
     11         self.session.get_api_list()

<...>synology_api\auth.py in login(self, application)
     40         else:
     41             session_request = requests.get(self._base_url + login_api, param, verify=self._verify)
---> 42             self._sid = session_request.json()['data']['sid']
     43             self._session_expire = False
     44             if self._debug is True:

KeyError: 'data'
```

The error is misleading, since `auth.login()` has no error checking on the `requests.get()` result, but what happens is that `session_request .json()` has this content:

```python
{'error': {'code': 402}, 'success': False}
```

And finally, `402` seems to be permission denied.

My first suggestion would be to improve the `auth.login()`in a similar fashion to `auth.logou()` and check for 
```python
if response.json()['success'] is True:
```

But there is also a PR which tries to improve the situation: https://github.com/N4S4/synology-api/pull/100
I like the addition of the error codes, but since the original author was not responding I wonder if it makes sense to make a similar PR which resolves your requests?

**Comment by eddybl** [comment]
I looked into the behavior of the `Authentication` class a bit more and I think there are some more general issues:

1. Failed login attempts might only result in an unrelated error message (see above)
2. The returns for `login()` etc are not used anywhere, since the returns are never used. This should probably be prints instead (for debugging)
3. most other methods other then `login()` seem to use a fixed `version: `, although they should probably all use `version: self._version`
4. A failed login does not result in any exception, so the code continues and may result in the debug output `You are now logged in!`

**Comment by N4S4** [comment]
It surely needs to improve, I was working on error handling function as PR #100 suggest but I wasn't much into it.
Ill try to work on it and take your suggestion time permitting

**Comment by eddybl** [comment]
I will try to work on a pull request. Once I started looking into it it got just bigger then I originally thought, so it might take a couple of days to find enough time ;)

**Comment by N4S4** [comment]
lol me too, however I'll do my best to improve this part

**Comment by N4S4** [comment]
ok so based on PR #100 and very little changes I managed to let error codes work and implemented in all modules, I still need to  add it for logout process.
I can upload to a new branck and if you want to test too would be great

**Comment by eddybl** [comment]
Ok, I will try to have a look this evening!

**Comment by eddybl** [comment]
I added a PR on the error_code branch: #108

**Comment by N4S4** [comment]
closed with PR #109
